prow/test/integration/README.md

```diff
@@ -8,14 +8,35 @@ Just run:
 ./prow/test/integration/integration-test.sh
 ```
 
-This script [sets up](#setup) the environment, [runs](#run-test) all the available integration tests, and then [cleans up](#cleanup) everything.
+This script creates a [KIND](https://kind.sigs.k8s.io/) Kubernetes cluster, runs all available integration tests, and finally deletes the cluster.
+
+# How it works
+
+Recall that Prow is a collection of services (Prow components) that can be
+deployed into a Kubernetes cluster. KIND provides an environment where we can
+deploy certain Prow components, and then from the integration tests we can
+create a Kubernetes Client to talk to this deployment of Prow.
+
+Note that the integration tests do not test all components (we need to fix
+this). [Here][test-cluster-images] is a list of components currently tested.
+These components are compiled and deployed into the test cluster on every
+invocation of [integration-test.sh](integration-test.sh).
+
+Each tested component needs a Kubernetes configuration so that KIND understands
+how to deploy it into the cluster, but that's about it (more on this below). The
+main thing to keep in mind is that the integration tests must be hermetic and
+reproducible. For this reason, all components that are tested must be configured
+so that they do not attempt to reach endpoints that are outside of the cluster.
+For example, this is why some Prow components have a `-github-endpoint=...` flag
+that you can use --- this way these components can be instructed to talk to the
+`fakeghserver` deployed inside the cluster instead of trying to talk to GitHub.
 
 ## Setup
```

Comment:
I think it would be nice to have a simple version of how to run these before getting into the nitty gritty. Ideally users will never have to run setup-local-registry.sh. They should just run ./prow/test/integration/integration-test.sh and let it take care of everything